tokio/src/sync/mpsc/bounded.rs

```diff
@@ -230,6 +230,74 @@ impl<T> Receiver<T> {
         poll_fn(|cx| self.chan.recv(cx)).await
     }
 
+    /// Receives the next values for this receiver and extends `buffer`.
+    ///
+    /// This method returns the number of values added to `buffer`.
+    ///
+    /// If there are no messages in the channel's queue, but the channel has
+    /// not yet been closed, this method will sleep until a message is sent or
+    /// the channel is closed.  Note that if [`close`] is called, but there are
+    /// still outstanding [`Permits`] from before it was closed, the channel is
+    /// not considered closed by `recv` until the permits are released.
+    ///
+    /// This method will never return 0 unless the channel has been closed and
+    /// there are no remaining messages in the channel's queue. This indicates
+    /// that no further values can ever be received from this `Receiver`. The
+    /// channel is closed when all senders have been dropped, or when [`close`]
+    /// is called.
+    ///
+    /// If at the time of the call `buffer` has unused capacity,
+    /// `recv_many` extends the buffer with no more elements than
+    /// its unused capacity. Should `buffer` have no initial unused
+    /// capacity, additional elements are first reserved.
+    ///
+    /// # Cancel safety
+    ///
+    /// This method is cancel safe. If `recv_many` is used as the event in a
+    /// [`tokio::select!`](crate::select) statement and some other branch
+    /// completes first, it is guaranteed that no messages were received on this
+    /// channel.
+    ///
+    /// [`close`]: Self::close
+    /// [`Permits`]: struct@crate::sync::mpsc::Permit///
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio::sync::mpsc;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let mut buffer: Vec<&str> = Vec::with_capacity(2);
+    ///     let (tx, mut rx) = mpsc::channel(100);
+    ///     {
+    ///         let tx2 = tx.clone();
+    ///         tx2.send("first").await.unwrap();
+    ///         tx2.send("second").await.unwrap();
+    ///         // Initial capacity ensures both values
+    ///         // can be added to the buffer.
+    ///         assert_eq!(2, rx.recv_many(&mut buffer).await);
+    ///         assert_eq!(vec!["first", "second"], buffer);
+    ///         tokio::spawn(async move {
+    ///             tx.send("third").await.unwrap();
+    ///         });
+    ///         // The 'tx' is dropped, but `recv_many`
+    ///         // is guaranteed not to return 0 as the channel
+    ///         // is not yet closed.  If the buffer is full, the next
+    ///         // call to `recv_many` reserves additional capacity.
+    ///         assert_eq!(1, rx.recv_many(&mut buffer).await);
+    ///         assert_eq!(vec!["first", "second", "third"], buffer);
+    ///     }
```

Comment:
Instead of a scope, please use `drop(tx)`. I think that is more clear.

Also, please add an empty line to visually separate each part of the test.